pkg/ocm/ocm.go

```diff
@@ -191,16 +203,58 @@ func (c *Client) GetResource(ctx context.Context, octx ocm.Context, cv *v1alpha1
 		return nil, "", fmt.Errorf("failed to resolve reference path to resource: %s %w", resource.Name, err)
 	}
 
-	access, err := res.AccessMethod()
-	if err != nil {
-		return nil, "", fmt.Errorf("failed to fetch access spec: %w", err)
-	}
+	// If resource is of type Helm we need to use a Downloader.
+	var (
+		mediaType string
+		reader    io.ReadCloser
+	)
+	// We add this decision because OCM is storing the Helm artifact as an ociArtifact at the
+	// time of this writing. This means, when fetching the resource via the normal route
+	// it will return an OCI blob instead of the actual helm chart content.
+	// Because of that, we need to create our own downloader when we are dealing with
+	// helm charts.
+	if res.Meta().Type == "helmChart" {
```

Comment:
maybe we can factor this out in a self contained function for responsibility separation concerns?

Reply:
Sure 👍